--- glusterfs-8.2/configure.ac.orig	2020-10-21 01:44:59.465980869 -0400
+++ glusterfs-8.2/configure.ac	2020-10-21 01:52:33.600980869 -0400
@@ -416,7 +416,7 @@
             [systemddir=$withval],
             [systemddir='${prefix}/lib/systemd/system'])
 AC_SUBST(systemddir)
-AM_CONDITIONAL([USE_SYSTEMD], test [ -d '/usr/lib/systemd/system' ])
+AM_CONDITIONAL([USE_SYSTEMD], test [ -d '/usr/lib/systemd' ])
 
 AC_ARG_WITH(initdir,
             [  --with-initdir=DIR init.d scripts in DIR @<:@/etc/init.d@:>@],
