--- glusterfs-8.0/rpc/rpc-lib/src/Makefile.am.orig	2020-07-22 08:09:31.315980869 -0400
+++ glusterfs-8.0/rpc/rpc-lib/src/Makefile.am	2020-07-22 08:10:12.203980869 -0400
@@ -2,8 +2,7 @@
 
 libgfrpc_la_SOURCES = auth-unix.c rpcsvc-auth.c rpcsvc.c auth-null.c \
 	rpc-transport.c xdr-rpc.c xdr-rpcclnt.c rpc-clnt.c auth-glusterfs.c \
-	rpc-drc.c $(CONTRIBDIR)/sunrpc/xdr_sizeof.c  rpc-clnt-ping.c \
-        autoscale-threads.c mgmt-pmap.c
+	rpc-drc.c rpc-clnt-ping.c autoscale-threads.c mgmt-pmap.c
 
 EXTRA_DIST = libgfrpc.sym
 
--- glusterfs-8.2/glusterfsd/src/Makefile.am.orig	2020-09-21 20:09:27.612000000 +0000
+++ glusterfs-8.2/glusterfsd/src/Makefile.am	2020-09-21 20:23:55.192000000 +0000
@@ -14,6 +14,7 @@
 		    $(top_builddir)/api/src/libgfapi.la \
 		    $(top_builddir)/rpc/rpc-lib/src/libgfrpc.la \
 		    $(top_builddir)/rpc/xdr/src/libgfxdr.la
+gf_attach_LDFLAGS = $(GF_LDFLAGS)
 
 noinst_HEADERS = glusterfsd.h glusterfsd-mem-types.h glusterfsd-messages.h
 
--- glusterfs-8.2/configure.ac.orig	2020-09-22 11:27:02.408000000 +0000
+++ glusterfs-8.2/configure.ac	2020-09-22 11:35:39.652000000 +0000
@@ -1491,9 +1491,9 @@
     ;;
 esac
 dnl GF_XLATOR_DEFAULT_LDFLAGS is for most xlators that expose a common set of symbols
-GF_XLATOR_DEFAULT_LDFLAGS='-avoid-version -export-symbols $(top_srcdir)/xlators/xlator.sym $(UUID_LIBS) $(GF_NO_UNDEFINED)'
+GF_XLATOR_DEFAULT_LDFLAGS='-avoid-version -export-symbols $(top_srcdir)/xlators/xlator.sym $(UUID_LIBS) $(GF_NO_UNDEFINED) $(TIRPC_LIBS)'
 dnl GF_XLATOR_LDFLAGS is for xlators that expose extra symbols, e.g. dht
-GF_XLATOR_LDFLAGS='-avoid-version $(UUID_LIBS) $(GF_NO_UNDEFINED)'
+GF_XLATOR_LDFLAGS='-avoid-version $(UUID_LIBS) $(GF_NO_UNDEFINED) $(TIRPC_LIBS)'
 
 AC_SUBST(GF_HOST_OS)
 AC_SUBST(GF_CFLAGS)
