--- glusterfs-8.0/libglusterd/src/Makefile.am.orig	2020-07-24 11:45:23.084000000 +0000
+++ glusterfs-8.0/libglusterd/src/Makefile.am	2020-07-24 15:47:54.108000000 +0000
@@ -12,7 +12,10 @@
 	-I$(top_srcdir)/rpc/xdr/src -I$(top_builddir)/rpc/xdr/src \
 	-I$(top_srcdir)/rpc/rpc-lib/src/
 
-libglusterd_la_LIBADD = $(ZLIB_LIBS) $(MATH_LIB) $(UUID_LIBS)
+
+libglusterd_la_LIBADD = $(ZLIB_LIBS) $(MATH_LIB) $(UUID_LIBS) \
+        $(shell ls /lib*/ld-linux-*.so.*)
+
 libglusterd_la_LDFLAGS = -version-info $(LIBGLUSTERFS_LT_VERSION) $(GF_LDFLAGS) \
 	-export-symbols $(top_srcdir)/libglusterd/src/libglusterd.sym
 
