--- glusterfs-coreutils-0.2.0/configure.ac.orig	2017-01-28 08:58:00.168000000 -0500
+++ glusterfs-coreutils-0.2.0/configure.ac	2017-01-28 09:14:39.792000000 -0500
@@ -1,7 +1,7 @@
 AC_PREREQ([2.69])
 
 AC_INIT([glusterfs-coreutils],
-        m4_esyscmd(echo -n `git describe --always --tags`),
+        m4_esyscmd(echo -n 0.2.0),
         [gluster-users@gluster.org],,
         [https://github.com/gluster/gluster-coreutils.git])
 
