--- glusterfs-3.7.17/configure.ac.orig	2016-11-04 08:16:19.652000000 -0400
+++ glusterfs-3.7.17/configure.ac	2016-11-04 08:18:06.908000000 -0400
@@ -1189,8 +1189,8 @@
     dnl Find python libs at user configured libdir and also "lib" under prefix
     PYTHONDEV_LDFLAGS="${PYTHON_LIBS} -L`${PYTHON}-config --prefix`/lib -L`${PYTHON}-config --prefix`/$libdir `${PYTHON}-config --ldflags`"
 
-    BUILD_PYTHON_SITE_PACKAGES=`$PYTHON -c 'from distutils.sysconfig import get_python_lib; print(get_python_lib(prefix="${exec_prefix}"))'`
-    BUILD_PYTHON_INC=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_inc()"`
+    BUILD_PYTHON_SITE_PACKAGES=${pythondir}
+    BUILD_PYTHON_INC=`$PYTHON -c "import sys; from distutils import sysconfig; sys.stdout.write(sysconfig.get_python_inc())" 2>/dev/null`
     BUILD_PYTHON_LIB=python$PYTHON_VERSION
 
     GLUPY_SUBDIR=glupy
