--- glusterfs-3.9.0/configure.ac.orig	2016-11-15 14:40:38.956000000 -0500
+++ glusterfs-3.9.0/configure.ac	2016-11-15 14:42:17.332000000 -0500
@@ -201,6 +201,7 @@
                 extras/init.d/glusterd-Redhat
                 extras/init.d/glusterd-FreeBSD
                 extras/init.d/glusterd-SuSE
+                extras/init.d/glustereventsd-Debian
                 extras/init.d/glustereventsd-Redhat
                 extras/init.d/glustereventsd-FreeBSD
                 extras/ganesha/Makefile
@@ -516,7 +517,7 @@
 
 dnl Check if version matches that we require
 PYTHONDEV_CPPFLAGS=
-PYTHONDEV_CPPFLAGS=
+PYTHONDEV_LDFLAGS=
 BUILD_PYTHON_SITE_PACKAGES=
 BUILD_PYTHON_INC=
 BUILD_PYTHON_LIB=
@@ -537,8 +538,8 @@
    dnl Find python libs at user configured libdir and also "lib" under prefix
    PYTHONDEV_LDFLAGS="${PYTHON_LIBS} -L`${PYTHON}-config --prefix`/lib -L`${PYTHON}-config --prefix`/$libdir `${PYTHON}-config --ldflags`"
 
-   BUILD_PYTHON_SITE_PACKAGES=`$PYTHON -c 'from distutils.sysconfig import get_python_lib; print(get_python_lib(prefix="${exec_prefix}"))'`
-   BUILD_PYTHON_INC=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_inc()"`
+   BUILD_PYTHON_SITE_PACKAGES=${pythondir}
+   BUILD_PYTHON_INC=`$PYTHON -c "import sys; from distutils import sysconfig; sys.stdout.write(sysconfig.get_python_inc())" 2>/dev/null`
    BUILD_PYTHON_LIB=python$PYTHON_VERSION
 
    dnl Now check for python header using the include path obtained above
--- glusterfs-3.9.0/extras/init.d/Makefile.am.orig	2016-11-15 09:54:27.289711290 -0500
+++ glusterfs-3.9.0/extras/init.d/Makefile.am	2016-11-15 11:44:46.960000000 -0500
@@ -1,7 +1,7 @@
 
 EXTRA_DIST = glusterd-Debian glusterd-FreeBSD glusterd-Redhat glusterd-SuSE \
 	glusterd.plist rhel5-load-fuse.modules \
-	glustereventsd-FreeBSD glustereventsd-Redhat
+	glustereventsd-FreeBSD glustereventsd-Redhat glustereventsd-Debian
 
 CLEANFILES =
 
--- glusterfs-3.9.0/cli/src/cli.c.orig	2016-11-15 13:37:08.504000000 -0500
+++ glusterfs-3.9.0/cli/src/cli.c	2016-11-15 13:40:18.100000000 -0500
@@ -66,9 +66,9 @@
 /* using argp for command line parsing */
 
 const char *argp_program_version = ""                                 \
-        PACKAGE_NAME" "PACKAGE_VERSION" built on "__DATE__" "__TIME__ \
+        PACKAGE_NAME" "PACKAGE_VERSION"\n" \
         "\nRepository revision: " GLUSTERFS_REPOSITORY_REVISION "\n"  \
-        "Copyright (c) 2006-2011 Gluster Inc. "                       \
+        "Copyright (c) 2006-2016 Gluster Inc. "                       \
         "<http://www.gluster.com>\n"                                  \
         "GlusterFS comes with ABSOLUTELY NO WARRANTY.\n"              \
         "You may redistribute copies of GlusterFS under the terms of "\
--- glusterfs-3.9.0/glusterfsd/src/glusterfsd.c.orig	2016-11-15 13:39:15.792000000 -0500
+++ glusterfs-3.9.0/glusterfsd/src/glusterfsd.c	2016-11-15 13:40:01.156000000 -0500
@@ -88,9 +88,9 @@
 static char argp_doc[] = "--volfile-server=SERVER [MOUNT-POINT]\n"       \
         "--volfile=VOLFILE [MOUNT-POINT]";
 const char *argp_program_version = ""
-        PACKAGE_NAME" "PACKAGE_VERSION" built on "__DATE__" "__TIME__
+        PACKAGE_NAME" "PACKAGE_VERSION"\n"
         "\nRepository revision: " GLUSTERFS_REPOSITORY_REVISION "\n"
-        "Copyright (c) 2006-2013 Red Hat, Inc. <http://www.redhat.com/>\n"
+        "Copyright (c) 2006-2016 Red Hat, Inc. <http://www.redhat.com/>\n"
         "GlusterFS comes with ABSOLUTELY NO WARRANTY.\n"
         "It is licensed to you under your choice of the GNU Lesser\n"
         "General Public License, version 3 or any later version (LGPLv3\n"
--- a/api/src/gfapi.aliases
+++ b/api/src/gfapi.aliases
@@ -141,6 +141,7 @@
 
 _pub_glfs_h_lookupat _glfs_h_lookupat$GFAPI_3.7.4
 
+_pub_glfs_free _glfs_free$GFAPI_3.7.16
 _pub_glfs_h_poll_upcall _glfs_h_poll_upcall$GFAPI_3.7.16
 _pub_glfs_upcall_get_fs _glfs_upcall_get_fs$GFAPI_3.7.16
 _pub_glfs_upcall_get_reason _glfs_upcall_get_reason$GFAPI_3.7.16
--- a/api/src/gfapi.map
+++ b/api/src/gfapi.map
@@ -175,6 +175,7 @@
 
 GFAPI_3.7.16 {
 	global:
+		glfs_free;
 		glfs_upcall_get_fs;
 		glfs_upcall_get_reason;
 		glfs_upcall_inode_get_event;
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -769,6 +769,9 @@
 glfs_fd_t *glfs_dup (glfs_fd_t *fd) __THROW
         GFAPI_PUBLIC(glfs_dup, 3.4.0);
 
+void glfs_free (void *ptr) __THROW
+        GFAPI_PUBLIC(glfs_free, 3.7.16);
+
 /*
  * Nobody needs this call at all yet except for the test script.
  */
--- /dev/null
+++ b/extras/init.d/glustereventsd-Debian.in
@@ -0,0 +1,91 @@
+#!/bin/sh
+### BEGIN INIT INFO
+# Provides:          glustereventsd
+# Required-Start:    $local_fs $network
+# Required-Stop:     $local_fs $network
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Gluster Events Server
+# Description:       Gluster Events Server
+### END INIT INFO
+
+# Author: Chris AtLee <chris@atlee.ca>
+# Patched by: Matthias Albert < matthias@linux4experts.de>
+
+PATH=/sbin:/usr/sbin:/bin:/usr/bin
+NAME=glustereventsd
+SCRIPTNAME=/etc/init.d/$NAME
+DAEMON=@prefix@/sbin/$NAME
+PIDFILE=/var/run/$NAME.pid
+GLUSTEREVENTSD_OPTS=""
+PID=`test -f $PIDFILE && cat $PIDFILE`
+
+
+# Gracefully exit if the package has been removed.
+test -x $DAEMON || exit 0
+
+# Load the VERBOSE setting and other rcS variables
+. /lib/init/vars.sh
+
+# Define LSB log_* functions.
+. /lib/lsb/init-functions
+
+
+do_start()
+{
+    pidofproc -p $PIDFILE $DAEMON >/dev/null
+    status=$?
+    if [ $status -eq 0 ]; then
+      log_success_msg "glustereventsd service is already running with pid $PID"
+    else
+      log_daemon_msg "Starting glustereventsd service" "glustereventsd"
+      start-stop-daemon --start --quiet --oknodo --pidfile $PIDFILE --startas $DAEMON -- -p $PIDFILE $GLUSTEREVENTSD_OPTS
+      log_end_msg $?
+      start_daemon -p $PIDFILE $DAEMON -f $CONFIGFILE
+      return $?
+    fi
+}
+
+do_stop()
+{
+    log_daemon_msg "Stopping glustereventsd service" "glustereventsd"
+    start-stop-daemon --stop --quiet --oknodo --pidfile $PIDFILE
+    log_end_msg $?
+    rm -f $PIDFILE
+    killproc -p $PIDFILE $DAEMON
+    return $?
+}
+
+do_status()
+{
+     pidofproc -p $PIDFILE $DAEMON >/dev/null
+     status=$?
+     if [ $status -eq 0 ]; then
+       log_success_msg "glustereventsd service is running with pid $PID"
+     else
+       log_failure_msg "glustereventsd service is not running."
+     fi
+     exit $status
+}
+
+case "$1" in
+  start)
+        do_start
+        ;;
+  stop)
+        do_stop
+        ;;
+  status)
+        do_status;
+        ;;
+  restart|force-reload)
+        do_stop
+        sleep 2
+        do_start
+        ;;
+  *)
+        echo "Usage: $SCRIPTNAME {start|stop|status|restart|force-reload}" >&2
+        exit 3
+        ;;
+esac
+
