--- nfs-ganesha-2.3.2/src/include/nfsv41.h.org	2016-05-03 18:10:18.701170555 +0530
+++ nfs-ganesha-2.3.2/src/include/nfsv41.h	2016-05-03 18:10:35.175228434 +0530
@@ -3646,7 +3646,7 @@
 				return false;
 		/* skip any further elements and lie on bitmap len */
 		for (i = mapsize; i < objp->bitmap4_len; i++) {
-			u_int crud;
+			u_int crud = 0;
 
 			if (!inline_xdr_u_int32_t(xdrs, &crud))
 				return false;
